common/djangoapps/student/models.py

```
@@ -1310,7 +1311,20 @@ def enroll(cls, user, course_key, mode=None, check_access=False):
         verified the user authentication.
 
         Also emits relevant events for analytics purposes.
+
+        This method triggers a public signal that may prevent the creation of the enrollment.
         """
+
+        # This signal allows plugins to extend the behaviour of the enrollment process.
+        hook_result = pre_enrollment.send(sender=None, user=user, course_key=course_key, mode=mode)
```

how would look like the `result` using the current receiver in eox-hooks?

---

If you want to stop the enrollment from happening, you need to raise an exception of any kind